app/Commands/NewCommand.php

```diff
@@ -0,0 +1,474 @@
+
+        $checks = ['yarn', 'npm', 'git', 'valet', 'laravel', 'composer'];
+        foreach ($checks as $check) {
+            $this->tools[$check] = $finder->find($check);
+        }
+    }
+
+    protected function setBasePath()
+    {
+        $this->basepath = $this->cwd;
+        if ($this->option('path')) {
+            $path = $this->option('path');
+            if (is_dir($path)) {
+                $this->basepath = $path;
+            } else {
+                $this->warn("Your defined '--path $path' is not a directory, so I am skipping it and using '{$this->basepath}' instead.");
+            }
+        }
+
+        return $this->basepath;
+    }
+
+    protected function askToAndRemoveProject()
+    {
+        $this->info("The directory '{$this->projectpath}' already exists.");
+
+        if ($this->confirm("Shall I proceed by removing the following directory? {$this->projectpath}")) {
+            $fs = new Filesystem\Filesystem();
+
+            // @todo Use laravel --force here instead of deleteDirectory()?
+            if ($fs->deleteDirectory($this->projectpath)) {
+                $this->info("I removed the following directory: {$this->projectpath}");
+
+                return true;
+            } else {
+                $this->error("I was unable to remove the '{$this->projectpath}' directory so I must exit.");
+
+                return false;
+            }
+        } else {
+            $this->error("You have chosen to not remove the '{$this->projectpath}' directory so I must exit.");
+
+            return false;
+        }
+    }
+
+    protected function replaceEnvVariables()
+    {
+        // @todo make this a configuration option
+        $changes = [
+            'DB_DATABASE' => $this->projectname,
+            'DB_USERNAME' => 'root',
+            'DB_PASSWORD' => '',
+            'APP_URL' => $this->projecturl,
+        ];
+
+        // @todo There has to be a better way; opening/closing a file 4x is wrong
+        foreach ($changes as $name => $newvalue) {
+            $this->replaceEnvVariable($name, $newvalue);
+        }
+    }
+
+    protected function replaceEnvVariable($name, $newvalue)
+    {
+        // @todo will .env always exist here? Check if not and copy over .env.example?
+        // @todo research a more official way to replace .env values... I could not find one
+        $contents = file_get_contents($this->projectpath.DIRECTORY_SEPARATOR.'.env');
+
+        $newcontents = $contents;
+        preg_match("@$name=(.*)@", $contents, $matches);
+        if (isset($matches[1])) {
+            // @todo sanitize new value and research .env guidelines
+            $newcontents = str_replace("$name=$matches[1]", "$name=$newvalue", $newcontents);
+        }
+
+        file_put_contents($this->projectpath.DIRECTORY_SEPARATOR.'.env', $newcontents);
+
+        return ($newcontents !== $contents) ? true : false;
+    }
+
+    protected function doNodeOrYarn()
+    {
+        $command = '';
```

Comment:
I'd start it `null` instead of empty string. A bit clearer.

Reply:
Fixed.